CIP-0109/README.md

```diff
@@ -0,0 +1,97 @@
+
+
+$$
+\begin{equation}
+a \equiv \underbrace{(b \times b \times ... \times b)}_{e \textrm{ times}} (\bmod{m} )
+\end{equation}
+$$
+
+That said, in the context of multiplicative inversion, a negative exponent can be interpreted as taking the exponent of the inverse of the base number. That is
+
+$$
+\begin{equation}
+b^{-e} (\bmod{m}) := (b^{-1})^{e} (\bmod{m})
+\end{equation}
+$$
+
+We propose to also include this extension to the plutus built-in, for optimized inversion when this is possible. This is inversion is not guaranteed to exist for all numbers $b$, only when the modulus $m$ is a prime number this is guaranteed. We also propose that the built-in fails if this inverse does not exist, and if a modulus is provided that is smaller than one.
+
+### Function definition
+With the above, we define a new Plutus built-in function with the following type signature
+
+```hs
+modularExponentiation :: Integer -> Integer -> Integer -> Integer
+```
+here the first argument is the base, the second the exponent and the third the modulus. As mentioned above, the behavior of this function is that it fails if the modulus is not a positive integer, or if the inverse of the base does not exist for a negative exponent. For the lower level implementation, we propose the usage of the `integerPowMod` function in the `ghc-bignum` packages. This function has the desired functionality, is optimized, and is easy to integrate in the plutus stack.
+
+### Cost model
+The computational impact of modular exponentiation is complexified by it having three arguments. That said, observe that the integers used can always be bound by the modulus. Preliminary [benchmarks](https://github.com/perturbing/expFast-bench/tree/cca69b842050de9523493d52c20384bc50c80b22) on the time consumption of this `integerPowMod` function show that it can be costed constant in the size of its first argument (the base) and linear in the other two.
+
+## Rationale: how does this CIP achieve its goals?
+Integrating this function directly into Plutus will streamline cryptographic operations, reduce transaction costs, and uphold the integrity of existing cryptographic interfaces. It addresses current inefficiencies and enhances the cryptographic capabilities of the Plutus platform.
+
+## Path to Active
+
+### Acceptance Criteria
+
+We consider the following criteria to be essential for acceptance:
+
+- [ ] The PR for this functionality is merged in the Plutus repository.
+- [ ] This PR must include tests, demonstrating that it behaves as the specification requires in this CIP.
+- [ ] A benchmarked use case is implemented in the Plutus repository, demonstrating that realistic use of this primitive does, in fact, provide major cost savings.
+
+### Implementation Plan
+
+IOG has begun the implementation as required in the acceptance criteria.
```

Comment:
```suggestion
- [x] IOG Plutus team consulted and accept the proposal.
- [x] Author to provide preliminary benchmarks of time consumption.
  - https://github.com/perturbing/expFast-bench/tree/cca69b842050de9523493d52c20384bc50c80b22
```

Just to fill this in

Reply:
@perturbing @iquerejeta @michaelpj @kwxm - this is a standard level of detail, so if incorrect please let us know in the next day or two since this passed `Last Check` at our CIP meeting today & we're going to merge it & would commit the above unless any objections.